### ISSUE TITLE
add mac support

### ISSUE THREAD

**Issue by MrWijnen01** [ECTO-1A/AppleJuice#8]
would be nice if there is support for mac

**Comment by lkruesselmann** [comment]
yes please!

**Comment by chris01b** [comment]
This project requires the [BlueZ](http://www.bluez.org/) Linux Bluetooth protocol stack which is built for the linux kernel. Mac uses a higher-level [Core Bluetooth](https://developer.apple.com/documentation/corebluetooth) and [IOBluetooth](https://developer.apple.com/documentation/iobluetooth) framework to gain user-space access to Bluetooth devices. They are considerably different. It's still worth investigating if it could be done though.

**Comment by chris01b** [comment]
It would be worth trying this out with virtualization and a usb bluetooth dongle and sharing that usb bus. I set it up with Docker running the latest Ubuntu on my Intel Macbook but the built-in bluetooth was not listed under `hcitool dev`.

This was my Dockerfile:
```
# Use the official Ubuntu image as a base image
FROM ubuntu:latest

# Set the working directory in the container
WORKDIR /AppleJuice

# Install necessary packages and clone the repository
RUN apt-get update && \
    apt-get install -y bluez libcap2-bin libpcap-dev libev-dev libnl-3-dev libnl-genl-3-dev libnl-route-3-dev cmake libbluetooth-dev git python3-pip && \
    git clone https://github.com/ECTO-1A/AppleJuice.git /AppleJuice

# Install Python dependencies
COPY requirements.txt /AppleJuice/
RUN pip3 install git+https://github.com/pybluez/pybluez.git#egg=pybluez pycryptodome
RUN pip3 install --no-cache-dir -r /AppleJuice/requirements.txt

# Install Python packages listed in requirements.txt
COPY requirements.txt /AppleJuice/
RUN pip3 install --no-cache-dir -r /AppleJuice/requirements.txt

# Set capabilities to allow the Python binary to access raw sockets
RUN setcap cap_net_raw,cap_net_admin+eip $(eval readlink -f $(which python3))
```

**Comment by ECTO-1A** [comment]
Yeah I tested using Kali in virtual box on my intel Mac and couldn't get it to pass through the USB Bluetooth adapter. There were a few "tricks" people suggested but none work. It seems like once the Mac attaches to it, it won't pass through and I still haven't found a workaround. If anyone has suggestions for me to test out please let me know!

**Comment by JamieGrimwood** [comment]
Can't get this working either, even side loading Kali linux on my mac and booting it that way, I struggled to even get the wifi drivers installed let alone the bluetooth ones. Really hope MacOS support comes soon!

**Comment by huajip** [comment]
I've been running both Ubuntu and Kali Linux virtual machines on my M1 MacBook using Parallels Desktop. In Ubuntu, the "Bluetooth sharing with Linux" feature seems to work well, and I can successfully identify my Bluetooth devices using the hcitool dev command. However, even after identifying the devices and executing the necessary scripts, there's no response on my iPhone. On the other hand, Kali Linux seems to lack this support entirely; running the hcitool dev command returns an empty list of devices.

**Comment by JamieGrimwood** [comment]
> I've been running both Ubuntu and Kali Linux virtual machines on my M1 MacBook using Parallels Desktop. In Ubuntu, the "Bluetooth sharing with Linux" feature seems to work well, and I can successfully identify my Bluetooth devices using the hcitool dev command. However, even after identifying the devices and executing the necessary scripts, there's no response on my iPhone. On the other hand, Kali Linux seems to lack this support entirely; running the hcitool dev command returns an empty list of devices.

Yeah, Kali returned no devices on both a VM and dual booted on my Mac. I'll give Ubuntu a go but I've ordered a USB dongle now which will arrive tomorrow, so fingers crossed it works 😬

**Comment by liquidz00** [comment]
@JamieGrimwood any luck or still no dice?

**Comment by JamieGrimwood** [comment]
> @JamieGrimwood any luck or still no dice?

Coundn't get the dongle to show up sadly! Real shame as this would have been really cool software to play around with :(
